src/binary-search-tree.js

```diff
@@ -32,7 +70,25 @@ class BinarySearchTree {
   // again. Whatever floats your boat.
   /* eslint-disable global-require */
   breadthFirstForEach(cb) {
-
+    const q = new Queue();
```

Comment:
I'm not entirely sure why you're instantiating two queues. It seems like the logic behind your implementation is a bit convoluted. Perhaps think about what aspects of it are actually necessary, and what can you get rid of so that your solution is simpler.